JellyDust.Tests/Tests.cs

```diff
@@ -0,0 +1,165 @@
+                var c = sut.Connection;
+            }
+
+            A.CallTo(() => _connectionFactory.CreateOpenConnection()).MustNotHaveHappened();
+            A.CallTo(() => _transactionFactory.OpenTransaction(A<IDbConnection>.Ignored)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public void UsingAConnectionCreatesAnActualConnection()
+        {
+            UnitOfWork sut;
+            using (sut = new UnitOfWork(_transactionFactory, _connectionFactory))
+            {
+                var c = sut.Connection.DbConnection;
+            }
+
+            A.CallTo(() => _connectionFactory.CreateOpenConnection()).MustHaveHappenedOnceExactly();
+            A.CallTo(() => _transactionFactory.OpenTransaction(A<IDbConnection>.Ignored)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public void AskingForALazyTransactionDoesntDoAnythingEither()
+        {
+            UnitOfWork sut;
+            using (sut = new UnitOfWork(_transactionFactory, _connectionFactory))
+            {
+                var c = sut.Transaction;
+            }
+
+            A.CallTo(() => _connectionFactory.CreateOpenConnection()).MustNotHaveHappened();
+            A.CallTo(() => _transactionFactory.OpenTransaction(A<IDbConnection>.Ignored)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public void UsingATransactionCreatesBothATransactionAndAConnection()
+        {
+            UnitOfWork sut;
+            using (sut = new UnitOfWork(_transactionFactory, _connectionFactory))
+            {
+                var c = sut.Transaction.DbTransaction;
+                Assert.AreSame(sut.Transaction.DbConnection, sut.Connection.DbConnection);
+            }
+
+            A.CallTo(() => _connectionFactory.CreateOpenConnection()).MustHaveHappenedOnceExactly();
+            A.CallTo(() => _transactionFactory.OpenTransaction(A<IDbConnection>.That.IsSameAs(_connection))).MustHaveHappenedOnceExactly();
+        }
+
+        [Test]
+        public void TestingSomeDisposing()
+        {
+            UnitOfWork sut;
+            using (sut = new UnitOfWork(_transactionFactory, _connectionFactory))
+            {
+                var c = sut.Transaction.DbTransaction;
+            }
+
+            A.CallTo(() => _connection.Dispose()).MustHaveHappenedOnceExactly();
+            A.CallTo(() => _transaction.Dispose()).MustHaveHappenedOnceExactly();
+        }
+
+        [Test]
+        public void OneTransactionPerUnitOfWorkWeSaid()
+        {
+            UnitOfWork sut;
+            using (sut = new UnitOfWork(_transactionFactory, _connectionFactory))
+            {
+                using (var t1 = sut.Transaction)
+                {
+                    var doSomething = t1.DbTransaction;
+                }
+
+                ITransaction t2;
+                Assert.Throws<InvalidOperationException>(() => t2 = sut.Transaction);
+            }
+        }
+
+
+        [Test]
+        public void SomeoneElseShouldBeAbleToUseOurConnectionThough()
```

Comment:
Is there a use case for this scenario?

Reply:
Hmm. I guess if you for some reason want to access the Transaction like in this test, then you explicitly tell when to create the Transaction, and you might want to explicitly dispose it as well.... Im just making sure you're not disposing the connection at the same time though. 

Ordinarily, the Transaction should be retrieved from the UnitOfWork instead of directly from the Connection